mentions/src/main/kotlin/io/spine/examples/pingh/mentions/GitHubClientProcess.kt

```diff
@@ -110,13 +111,30 @@ public class GitHubClientProcess :
         val userMentionedEvents = createUserMentionedEvents(mentions)
         val mentionsUpdateFromGitHubCompleted =
             MentionsUpdateFromGitHubCompleted::class.buildBy(state().id)
-        builder().clearWhenStarted()
+        builder()
+            .setWhenLastSuccessfulUpdate(state().whenStarted)
+            .clearWhenStarted()
         return userMentionedEvents
-            .toList()
+            .toList<EventMessage>()
             .plus(mentionsUpdateFromGitHubCompleted)
     }
 
-    private fun createUserMentionedEvents(gitHubMentions: Set<Mention>): Set<EventMessage> =
+    /**
+     * Defines and returns the time when the mentions were last updated.
+     *
+     * If updates were already performed, it returns the time of the last successfully
+     * completed update. If no updates are made, it returns the time equal to midnight
+     * of the previous work day.
+     */
+    private fun defineLastSuccessfulUpdate(): Timestamp {
```

Comment:
I don't understand the naming. "Definition" is something very different from what you do. Please consult a dictionary, and come up with something better. And, preferably, shorter in general.